core/server/data/migrations/versions/3.24/01-populate-group-for-new-portal-settings.js

```diff
@@ -0,0 +1,39 @@
+const logging = require('../../../../../shared/logging');
+
+// new setting keys and group mapping
+const groupMapping = [{
+    group: 'portal',
+    keys: [
+        'portal_button_style',
+        'portal_button_icon',
+        'portal_button_signup_text'
+    ]
+}];
+
+module.exports = {
+    config: {
+        transaction: true
+    },
+
+    async up(options) {
+        // set the correct group value for new settings
+        await Promise.map(groupMapping, async (groupMap) => {
+            return await Promise.map(groupMap.keys, async (key) => {
+                logging.info(`Updating setting ${key} to group ${groupMap.group}`);
+
+                return await options
```

Comment:
I feel like this whole `up` method could be simplified to
```js
await options
  .transacting('settings')
  .update({
    group: 'portal'
  })
  .whereIn('key', ['portal_button_style', 'portal_button_icon', 'portal_button_signup_text');
```

Reply:
@allouis Primary reason for structuring it like this is allowing future settings migration for group/flags to reuse this easily and just update the mapping on top

Reply:
@rishabhgrg I'm siding with @allouis on this one. You could still keep the mapping on top and then use something like:
```js
await options
  .transacting('settings')
  .update({
    group: groupMap.gorup
  })
  .whereIn('key', groupMap.keys);
```
At least that would limit the nesting to one level and keep the configuration more-less generic :thinking: